web/tests/search.test.cjs

```diff
@@ -128,6 +128,88 @@ run_test("initialize", ({override, override_rewire, mock_template}) => {
+            assert.equal(opts.item_html(source[1], "top"), expected_value);
+
+            /* Test sorter */
+            assert.equal(opts.sorter(search_suggestions.strings), search_suggestions.strings);
+
+            search_suggestions = {
+                lookup_table: new Map([
+                    [
+                        "topic:",
+                        {
+                            description_html: `Topic <span class="empty-topic-display">general chat</span>`,
+                            search_string: "topic:",
+                        },
+                    ],
+                ]),
+                strings: ["topic:"],
+            };
+
+            expected_source_value = search_suggestions.strings;
+            source = opts.source("topic:");
+            assert.deepStrictEqual(source, expected_source_value);
+
+            /* Test topic empty operand suggestion */
+            description_html = `Topic <span class="empty-topic-display">general chat</span>`;
+            search_string = `topic:<span class="empty-topic-display"> translated: general chat</span>`;
+            expected_value = `<div class="search_list_item">\n            <span class="pill-container"><div class='pill ' tabindex=0>\n    <span class="pill-label">\n        <span class="pill-value">\n            ${search_string}\n        </span></span>\n    <div class="exit">\n        <a role="button" class="zulip-icon zulip-icon-close pill-close-button"></a>\n    </div>\n</div>\n</span>\n            <div class="description">${description_html}</div>\n</div>\n`;
+            assert.equal(opts.item_html(source[0], "topic:"), expected_value);
+
+            search_suggestions = {
+                lookup_table: new Map([
+                    [
+                        "topic: zo",
+                        {
+                            description_html: undefined,
+                            search_string: "topic: zo",
+                        },
+                    ],
+                ]),
+                strings: ["topic: zo"],
+            };
+
+            /* Test topic empty operand suggestion */
+            source = opts.source("topic: zo");
+            expected_value = `<div class="search_list_item">\n            <span class="pill-container"><div class='pill ' tabindex=0>\n    <span class="pill-label">\n        <span class="pill-value">\n            topic:<span class="empty-topic-display"> translated: general chat</span>\n        </span></span>\n    <div class="exit">\n        <a role="button" class="zulip-icon zulip-icon-close pill-close-button"></a>\n    </div>\n</div>\n</span>\n                    <div class="description">search for zo</div>\n    \n</div>\n`;
```

Comment:
Hm, this test is leaving me wondering if we want e.g. "topic: zulip" to give search results of _general chat_ and "search for zulip", instead of topics with zulip in it. If you remove the space, you get topics with zulip in it. Maybe we should start a conversation about that on CZO. But we could merge these tests before that conversation gets figured out